pkg/api/v1alpha1/cloudstackdatacenterconfig_types.go

```diff
@@ -139,9 +160,34 @@ func (v *CloudStackDatacenterConfig) Marshallable() Marshallable {
 }
 
 func (v *CloudStackDatacenterConfig) Validate() error {
+	// TODO https://github.com/aws/eks-anywhere/issues/2406: Add validation to
+	// * Make sure that v.Spec.Zones, v.Spec.Domain, v.Spec.Account, v.Spec.ManagementApiEndpoint are all empty
+	// * Make sure that len(v.Spec.AvailabilityZones) > 0
+	// * Make sure that all the AZ names are unique
 	return nil
 }
 
+func (v *CloudStackDatacenterConfig) SetDefaults() {
+	if v.Spec.AvailabilityZones == nil || len(v.Spec.AvailabilityZones) == 0 {
+		v.Spec.AvailabilityZones = make([]CloudStackAvailabilityZone, 0, len(v.Spec.Zones))
+		for index, csZone := range v.Spec.Zones {
+			az := CloudStackAvailabilityZone{
+				Name:                  fmt.Sprintf("availability-zone-%d", index),
```

Comment:
this is subject to change pending alignment with CAPC for backwards compatibility